### PR TITLE
cleanup utils and add unit tests

### LINKED ISSUE #11: PR for utility functions
There exists utility functions around timestamp and relative delta handling. This code should be cleaned up and pushed to main after initial setup PR is merged.

https://github.com/SUSE-Enceladus/csp-billing-adapter/blob/devel/csp_billing_adapter/utils.py

### PR DESCRIPTION
**Pull request by KeithMnemonic**
This PR cleans up the utils function and also add unit tests

closes #11

### CODE REVIEW

File: csp_billing_adapter/utils.py [hunk 1]
```diff
@@ -13,29 +13,45 @@
...
+    """get current time"""
     return datetime.datetime.now(datetime.timezone.utc)
 
 
 def date_to_string(date):
-    return date.isoformat()
+    """convert date to string"""
+    try:
+        return date.isoformat()
+    except (AttributeError, ValueError) as exc:
```
Comment (rtamalin) [inline csp_billing_adapter/utils.py:31]: Probably should be logging something useful here given that we are catching and then re-raising the exceptions ;-)

File: csp_billing_adapter/utils.py [hunk 2]
```diff
@@ -13,29 +13,45 @@
...
+        raise exc
 
 
 def string_to_date(timestamp):
-    return datetime.datetime.fromisoformat(timestamp)
+    """convert string to date"""
+    try:
+        return datetime.datetime.fromisoformat(timestamp)
+    except ValueError as exc:
```
Comment (rtamalin) [inline csp_billing_adapter/utils.py:39]: Ditto

File: csp_billing_adapter/utils.py [hunk 3]
```diff
@@ -13,29 +13,45 @@
...
+        raise exc
 
 
 def get_date_delta(date: datetime.datetime, delta: int):
-    return date + datetime.timedelta(seconds=delta)
+    """get a new date using a provided date and delta offset"""
+    try:
+        return date + datetime.timedelta(seconds=delta)
+    except TypeError as exc:
```
Comment (rtamalin) [inline csp_billing_adapter/utils.py:47]: Ditto

**Review by rtamalin — APPROVED** [review]
LGTM - we can add the logging later as part of the logging updates.

File: csp_billing_adapter/utils.py (hunk 1 above)
Comment (rtamalin) [reply csp_billing_adapter/utils.py:31]: So my point really is that if we are catching and re-raising the same exception, but not doing anything with it, there is not much point in doing the exception catch/re-raise... Usually in such a case I would be logging that the exception occurred and any additional useful information that is available in the content where the exception was caught...

Comment (KeithMnemonic) [reply csp_billing_adapter/utils.py:31]: i will add logging to these. any objections?

**Review by brett060102 — APPROVED** [review]
LGTM

**Pull request merged**